resources/js/oauth2/profile/edit_client/edit_client.js

```diff
@@ -0,0 +1,297 @@
+import React, {useEffect, useState} from "react";
+import ReactDOM from "react-dom";
+import {MuiThemeProvider, createTheme} from "@material-ui/core/styles";
+import AssignmentIcon from "@material-ui/icons/Assignment";
+import CheckCircleIcon from "@material-ui/icons/CheckCircle";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import InfoOutlinedIcon from "@material-ui/icons/InfoOutlined";
+import Swal from "sweetalert2";
+import Navbar from "../../../components/navbar/navbar";
+import TopLogo from "../../../components/top_logo/top_logo";
+import OauthPanel from "./components/oauth_panel";
+import AllowedScopesPanel from "./components/allowed_scopes_panel";
+import AppGrantsPanel from "./components/app_grants_panel";
+import SecuritySettingsPanel from "./components/security_settings_panel";
+import {
+    Accordion,
+    AccordionDetails,
+    AccordionSummary,
+    Card,
+    CardContent,
+    Container,
+    CssBaseline,
+    Divider,
+    Grid,
+    Tooltip,
+    Typography
+} from "@material-ui/core";
+import {
+    addScope,
+    getAccessTokens,
+    getRefreshTokens,
+    regenerateClientSecret,
+    revokeToken,
+    removeScope,
+    updateClientData
+} from "./actions";
+
+import styles from "./edit_client.module.scss";
+
+const EditClientPage = (
+    {
+        appLogo,
+        appType,
+        appTypes,
+        canRequestRefreshTokens,
+        clientId,
+        clientName,
+        clientType,
+        clientTypes,
+        csrfToken,
```

Comment:
lets see if we could use redux thunk to start passing the CSRFTOKEN on a custom param on middleware
see
https://github.com/reduxjs/redux-thunk#injecting-a-custom-argument